pkg/monitor/cluster/serviceprincipalexpiration.go

```diff
@@ -0,0 +1,36 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	utilgraph "github.com/Azure/ARO-RP/pkg/util/graph"
+)
+
+/****************************************************
+	Monitor the Service Prinicpal exipiration date
+	Condition:
+	 - Check if SP is going to expire in next 30 days
+****************************************************/
+
+func (mon *Monitor) emitSpExpiration(ctx context.Context) error {
+	date, err := utilgraph.GetServicePrincipalExpiryByAppID(ctx, mon.spGraphClient, mon.oc.Properties.ServicePrincipalProfile.ClientID)
+	if err != nil {
+		return err
+	}
+
+	expiryDate, err := time.Parse("2006-01-02", *date) // Convert string date in Date Time Format
+	if err != nil {
+		return err
+	}
+
+	if int(time.Until(expiryDate).Hours()/24) <= 30 { // Alert if no. of days left on SP expiration is less than or equal to 30
+		mon.emitGauge("cluster.serviceprincipal.expiration", 1, map[string]string{
+			"expiry-date": *date,
```

Comment:
We use camelCase on our monitor field names.
```suggestion
			"expiryDate": *date,
```

Reply:
Done